spec/bolt_server/transport_app_spec.rb

```diff
@@ -38,288 +38,255 @@ def file_data(file)
     expect(last_response.status).to eq(200)
   end
 
-  context 'with ssh target', ssh: true do
-    let(:target) { conn_info('ssh') }
-    let(:path) { '/ssh/run_task' }
-    let(:echo_task) {
-      {
-        'name': 'sample::echo',
-        'metadata': {
-          'description': 'Echo a message',
-          'parameters': { 'message': 'Default message' }
-        },
-        files: [{
-          filename: "echo.sh",
-          sha256: "foo",
-          uri: { path: 'foo', params: { environment: 'foo' } }
-        }]
-      }
-    }
-
-    it 'errors if both password and private-key-content are present' do
-      body = {
-        'task': echo_task,
-        'target': {
-          'password': 'foo',
-          'private-key-content': 'content'
-        },
-        'parameters': { "message": "Hello!" }
-      }
-
-      post path, JSON.generate(body), 'CONTENT_TYPE' => 'text/json'
+  context 'when raising errors' do
+    it 'returns non-html 404 when the endpoint is not found' do
+      post '/ssh/run_tasksss', JSON.generate({}), 'CONTENT_TYPE' => 'text/json'
       expect(last_response).not_to be_ok
-      expect(last_response.status).to eq(400)
+      expect(last_response.status).to eq(404)
       result = JSON.parse(last_response.body)
-      regex = %r{The property '#/target' of type object matched more than one of the required schemas}
-      expect(result['details'].join).to match(regex)
+      expect(result['msg']).to eq("Could not find route /ssh/run_tasksss")
+      expect(result['kind']).to eq("boltserver/not-found")
     end
 
-    it 'fails if no authorization is present' do
-      body = {
-        'task': echo_task,
-        'target': {
-          'hostname': target[:host],
-          'user': target[:user],
-          'port': target[:port],
-          'host-key-check': false
-        },
-        'parameters': { "message": "Hello!" }
-      }
-
-      post path, JSON.generate(body), 'CONTENT_TYPE' => 'text/json'
+    it 'returns non-html 500 when the request times out' do
+      get '/500_error'
       expect(last_response).not_to be_ok
-      expect(last_response.status).to eq(400)
-      result = last_response.body
-      expect(result).to match(%r{The property '#/target' of type object did not match any of the required schemas})
+      expect(last_response.status).to eq(500)
+      result = JSON.parse(last_response.body)
+      expect(result['msg']).to eq('500: Unknown error: Unexpected error')
+      expect(result['kind']).to eq('boltserver/server-error')
+    end
+  end
+
+  describe 'transport routes' do
```

Comment:
My goal with moving these test cases to a "transport routes" section was to unit test the `/ssh/:action` and `/winrm/:action` logic without making the test content specific to one action. These tests no longer use the `:ssh` or `:winrm` tags because the functions they call have been mocked. FWIW, there are actual echo task executions for ssh and winrm further down, but if we'd rather actually run these in the docker/vagrant environment, I can take this back to the drawing board.

Reply:
Since there are executions of actual executions of ssh/winrm further down I think there's a low level of risk in your approach here. 👍